packages/typespec-client-generator-core/test/methods/lro.test.ts

```diff
@@ -1162,3 +1162,155 @@ describe("getLroMetadata", () => {
     );
   });
 });
+
+describe("@markAsLro decorator", () => {
+  let basicRunner: SdkTestRunner;
+
+  beforeEach(async () => {
+    basicRunner = await createSdkTestRunner({
+      emitterName: "@azure-tools/typespec-csharp",
+    });
+  });
+
+  it("should mark regular operation as LRO when decorated with @markAsLro", async () => {
+    await basicRunner.compile(`
+      @service
+      namespace TestService {
+        model DeploymentResult {
+          id: string;
+          status: "InProgress" | "Succeeded" | "Failed";
+          location?: string;
+        }
+
+        model DeploymentConfig {
+          name: string;
+          template: string;
+        }
+
+        @Azure.ClientGenerator.Core.Legacy.markAsLro
+        @route("/deployments/{deploymentId}")
+        @post
+        op startDeployment(
+          @path deploymentId: string,
+          @body config: DeploymentConfig
+        ): DeploymentResult;
+      }
+    `);
+
+    const methods = basicRunner.context.sdkPackage.clients[0].methods;
+    strictEqual(methods.length, 1);
+
+    const method = methods[0];
+    strictEqual(method.kind, "lro");
+    strictEqual(method.name, "startDeployment");
+
+    // Check LRO metadata exists
+    const metadata = method.lroMetadata;
```

Comment:
I doubt if the final result is correct.

Reply:
it has to be, since that's what we return for the final result. I'll add another check to the test though